src/status_im/ui/screens/offline_messaging_settings/events.cljs

```diff
@@ -16,6 +18,46 @@
                               {:dispatch [:logout]}
                               (accounts-events/update-settings (assoc-in settings [:wnode network] wnode))))))
 
+(handlers/register-handler-fx
+ ::save-wnode-transaction
+ (fn [{:keys [db] :as cofx} [_ wnode hash]]
+   (let [network     (ethereum/network->chain-keyword (:network db))
+         settings    (get-in db [:account/account :settings])
+         transaction (get-in db [:wallet :transactions hash])]
+     (when (seq transaction)
+       (handlers-macro/merge-fx cofx
+                                {:dispatch [::save-wnode wnode hash]}
+                                (accounts-events/update-settings
+                                  (assoc-in settings [:wnode-payment network wnode] hash)))))))
+
+(handlers/register-handler-fx
+ ::pay-wnode
+ (fn [{{:keys [web3] :as db} :db} [_ wnode]]
+   (let [network                         (ethereum/network->chain-keyword (:network db))
+         {:keys [address amount symbol]} (get-in db [:inbox/wnodes network wnode :payment])
+         contract                        (:address (tokens/symbol->token network symbol))
+         from                            (get-in db [:account/account :address])
+         gas                             (ethereum/estimate-gas symbol)
+         gas-price                       (money/->wei :gwei 5)]
+     (assoc-in db [:wallet :send-transaction :waiting-signal?] false)
+     (erc20/transfer web3
```

Comment:
you need to create and use custom fx here, something like
```clojure
{:db (assoc....)
 :erc20-transfer {:web3 web3 :contract contract....}}
```
and registered fx
```
(re-frame/reg-fx
 :erc20-transfer
 (fn [{:keys [web3 contract...]}]
   (erc20/transfer web3 ....
```

Reply:
btw you don't need this `(assoc-in db [:wallet :send-transaction :waiting-signal?] false)` anymore, because you're using signals, so it should be just 

```clojure
{:erc20-transfer {:web3 web3 :contract contract....}}
```